# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/musl/hotfix-linux-inline.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- musl-1.2.2/include/features.h.vanilla	2021-10-21 17:39:26.634632506 +0200
+++ musl-1.2.2/include/features.h	2021-10-21 17:41:59.760626400 +0200
@@ -37,4 +37,11 @@
 
 #define __REDIR(x,y) __typeof__(x) x __asm__(#y)
 
+/* Forces a function to be always inlined. */
+/* The Linux kernel defines __always_inline in stddef.h (283d7573), and
+   it conflicts with this definition.  Therefore undefine it first to
+   allow either header to be included first. */
+#undef __always_inline
+#define __always_inline __inline __attribute__ ((__always_inline__))
+
 #endif
